Philips.CodeAnalysis.MoqAnalyzers/MockObjectsMustCallExistingConstructorsAnalyzer.cs

```diff
@@ -123,6 +124,56 @@ private void AnalyzeNewObject(SyntaxNodeAnalysisContext context)
 			VerifyMockAttempt(context, mockedClass, objectCreationExpressionSyntax.ArgumentList, true);
 		}
 
+		private void AnalyzeVariableDeclaration(SyntaxNodeAnalysisContext context)
+		{
+			var variableDeclaration = (VariableDeclarationSyntax)context.Node;
+
+			// Check if the variable type is Mock<T>
+			TypeInfo typeInfo = context.SemanticModel.GetTypeInfo(variableDeclaration.Type);
```

Comment:
My understanding is that calling "context.SemanticModel" is slow, and therefore, we are supposed to check a string compare (e.g., on "Mock") and only loading the semantic model if the string compare checks out first. Thoughts?